components/org.wso2.carbon.identity.scim2.common/src/main/java/org/wso2/carbon/identity/scim2/common/impl/SCIMUserManager.java

```diff
@@ -1783,134 +1783,362 @@ private int handleStartIndexEqualsNULL(Integer startIndex) {
+     *
+     * @param domainName
+     * @param node
+     */
+    private String updateAttributeValue(String domainName, ExpressionNode node) throws BadRequestException {
+
+        String attributeName = node.getAttributeValue();
+        String filterOperation = node.getOperation();
+        String attributeValue = node.getValue();
+        String extractedDomain;
+
+        if (StringUtils.isNotEmpty(domainName)) {
+            if (isDomainNameEmbeddedInGroupAttributeValue(filterOperation, attributeName, attributeValue)) {
+                int indexOfDomainSeparator = attributeValue.indexOf(CarbonConstants.DOMAIN_SEPARATOR);
+                extractedDomain = attributeValue.substring(0, indexOfDomainSeparator).toUpperCase();
+
+                // Check whether the domain name is equal to the extracted domain name from attribute value.
+                if (StringUtils.isNotEmpty(extractedDomain) && !extractedDomain.equalsIgnoreCase(domainName)) {
+                    throw new BadRequestException(String.format(
+                            " Domain name: %s in the domain parameter does not match with the domain name: %s in "
+                                    + "search attribute value of %s claim.", domainName, extractedDomain,
+                            attributeName));
+                }
+
+                if (StringUtils.isEmpty(extractedDomain)) {
+                    return domainName + CarbonConstants.DOMAIN_SEPARATOR + attributeValue;
+                } else {
+                    return attributeValue;
+                }
+            } else {
+                return domainName + CarbonConstants.DOMAIN_SEPARATOR + attributeValue;
+            }
+        } else {
+            return attributeValue;
+        }
+    }
+
+    /**
+     * Method to verify whether there is a domain in the attribute value.
+     *
+     * @param filterOperation Operation of the expression node
+     * @param attributeName   Attribute name of the expression node
+     * @param attributeValue  Value of the expression node
+     * @return Whether there is a domain embedded to the attribute value
+     */
+    private boolean isDomainNameEmbeddedInGroupAttributeValue(String filterOperation, String attributeName,
+            String attributeValue) {
+
+        // Checks whether the domain separator is in the attribute value.
+        if (StringUtils.contains(attributeValue, CarbonConstants.DOMAIN_SEPARATOR)) {
+
+            // Checks whether the attribute name is display name or display uri.
+            if (StringUtils.equals(attributeName, SCIMConstants.GroupSchemaConstants.DISPLAY_NAME_URI) || StringUtils
+                    .equals(attributeName, SCIMConstants.GroupSchemaConstants.DISPLAY_URI)) {
+
+                // Checks whether the operator is equal to EQ, SW, EW, CO.
+                if (SCIMCommonConstants.EQ.equalsIgnoreCase(filterOperation) || SCIMCommonConstants.SW
```

Comment:
How we are handling what are the supported filters? if an unsupported filter comes what will be the result?

Reply:
Fixed: e0b27781f03c5dc319612f3d1924e70dce6c2f21